main.go

```diff
@@ -22,59 +55,232 @@ func main() {
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "load: %v\n", err)
+				os.Exit(1)
+			}
+			if packages.PrintErrors(pkgsInner) > 0 {
+				os.Exit(1)
+			}
+
+			for _, pkgInner := range pkgsInner {
+				for _, f := range pkgInner.Syntax {
+					ast.Inspect(f, func(node ast.Node) bool {
+						funcDecl, ok := node.(*ast.FuncDecl)
+						if !ok {
+							return true
+						}
+
+						// CRUD and CRUDThenPoll function must have receiver in Pandora SDK, otherwise continue the ast iteration.
+						if funcDecl.Recv == nil {
+							return true
+						}
+
+						funcDeclRecvList := funcDecl.Recv.List
+						if funcDeclRecvList != nil && len(funcDeclRecvList) > 0 {
+							funcDeclName := funcDecl.Name.Name
+							if !strings.Contains(funcDeclName, "Create") && !strings.Contains(funcDeclName, "Update") && !strings.Contains(funcDeclName, "Delete") {
+								return true
+							}
+
+							if funcDeclRecvList[len(funcDeclRecvList)-1] == nil {
+								return true
+							}
+
+							recvType, ok := funcDeclRecvList[len(funcDeclRecvList)-1].Type.(*ast.Ident)
+							if !ok {
+								return true
+							}
+
+							/*
+								1. Cannot only use func name Create/Update/Delete to identify the unique function but need to add its pkg and receiver.
+								(because there could be cases where unique receiver + func name exist in multiple pkgs, and they represent different services.)
+								2. Record which pkg + receiver has CRUD(ThenPoll) function
+							*/
+							funcScanCache[pkgInner.PkgPath+"."+recvType.Name] = append(funcScanCache[pkgInner.PkgPath+"."+recvType.Name], funcDeclName)
+						}
+						return false
+					})
 				}
-				future := futureType.Obj()
-				if future.Name() != "FutureAPI" || future.Pkg().Path() != "github.com/Azure/go-autorest/autorest/azure" {
-					return true
+			}
+
+			if len(funcScanCache[funcRecvType]) > 0 {
+				/* Here is the pkg logic this program leverages: [CRUD pkg] funcRecvType == [CRUDThenPoll Pkg] pkgInner.PkgPath+"."+recvType.Name
+				e.g. [CRUD pkg] funcRecvType: 			  github.com/hashicorp/go-azure-sdk/resource-manager/fluidrelay/2022-05-26/fluidrelayservers.FluidRelayServersClient
+				     [CRUDThenPoll Pkg] pkgInner.PkgPath: github.com/hashicorp/go-azure-sdk/resource-manager/fluidrelay/2022-05-26/fluidrelayservers
+					 [CRUDThenPoll Pkg] recvType.Name: 	  FluidRelayServersClient
+				*/
+
+				pollFuncList := funcScanCache[funcRecvType]
+				for _, v := range pollFuncList {
+					// Find the case that CRUD() is called but its pandora SDK also contains CRUDThenPoll()
+					if selExpr.Sel.Name+"ThenPoll" == v {
+
+						falseAlert := ""
+
+						// There could be false alerts because though CRUD() is called, and they have CRUDThenPoll counterparts, there are async handling right after the sync CRUD(),
+						// which makes the func call still correct. Add below workaround to label possible false alerts but there might be mis-labeled ones.
+						if len(asnmt.Lhs) > 1 {
```

Comment:
Is there an example of this?

Reply:
[Here ](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/nginx/nginx_configuration_resource.go#L305-L312)is an example where the Poll logic is handled by the developer rather than leveraging Pandora SDK CRUDThenPoll(). This is not regarded as a bug.

[Here ](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/serviceconnector/service_connector_app_service_resource.go#L228)is an example the developer "received the Poll signal" but did not handle the Poll function later. This is a bug.